ceno_emul/src/rv32im.rs

```diff
@@ -110,6 +105,37 @@ pub struct DecodedInstruction {
     opcode: u32,
 }
 
+#[derive(Clone, Copy, Debug)]
+pub struct DecodedInstruction {
+    pub kind: InsnKind,
```

Comment:
> you really feel the need to emulate Java and slap a getter in front of everything

I missed that one. That is not true and that would be silly.

There can be data containers (pub fields), in which case the module commits to the representation as simple & stable. That means other modules can write the fields directly with low risk of misunderstandings. If over time it becomes more complex then it's time to switch. Rarely the other way.

Yes it could have been that way. Whoever decided against at the time. It could be that way again if you want to spend the time.